R/edata_replace.R

```diff
@@ -185,46 +274,140 @@ edata_replace.metabData <- function(omicsData, x, y){
   # pull off the identifier column #
-  edata <- edata[, -which(colnames(edata)==edata_id)]
-
+  edata <- edata[,-which(colnames(edata) == edata_id)]
+  
   e_meta <- omicsData$e_meta
-
+  
   # get count of the number of values replaced #
-  if(is.na(x)){
+  if (is.na(x)) {
     truefalse <- is.na(edata)
-  }else{
-    truefalse <- edata==x
+  } else{
+    truefalse <- edata == x
   }
-  num_replaced <- sum(truefalse, na.rm=TRUE)
-
+  num_replaced <- sum(truefalse, na.rm = TRUE)
+  
   # apply vector_replace to the cols of edata (identifier column has been pulled off) #
-  edata_new <- apply(edata, 2, vector_replace, x=x, y=y)
-
+  edata_new <- apply(edata, 2, vector_replace, x = x, y = y)
+  
   # add the identifier column back #
-  edata_new <- data.frame(edata_id=feature_names, edata_new, check.names=check_names)
-
+  edata_new <-
+    data.frame(edata_id = feature_names, edata_new, check.names = check_names)
+  
   # replace e_data in omicsData with edata_new #
-  if(is.null(e_meta)){
+  if (is.null(e_meta)) {
     emeta_cname = NULL
-  }else{
+  } else{
     emeta_cname = attr(omicsData, "cnames")$emeta_cname
   }
-
+  
   # create an updated metabData object (look at as.metabData function to see how the attributes are accessed) #
-  updated_data <- as.lipidData(e_data = edata_new, f_data = omicsData$f_data, e_meta = omicsData$e_meta, edata_cname = edata_id, emeta_cname=emeta_cname, fdata_cname = attr(omicsData, "cnames")$fdata_cname, techrep_cname = attr(omicsData, "cnames")$techrep_cname, data_scale = attr(omicsData, "data_info")$data_scale, is_normalized = attr(omicsData, "data_info")$norm_info$is_normalized, data_types=attr(omicsData, "data_info")$data_types, check.names = check_names)
-
-  attributes(updated_data)$group_DF <- attributes(omicsData)$group_DF
+  updated_data <-
+    as.lipidData(
+      e_data = edata_new,
+      f_data = omicsData$f_data,
+      e_meta = omicsData$e_meta,
+      edata_cname = edata_id,
+      emeta_cname = emeta_cname,
+      fdata_cname = attr(omicsData, "cnames")$fdata_cname,
+      techrep_cname = attr(omicsData, "cnames")$techrep_cname,
+      data_scale = attr(omicsData, "data_info")$data_scale,
+      is_normalized = attr(omicsData, "data_info")$norm_info$is_normalized,
+      data_types = attr(omicsData, "data_info")$data_types,
+      check.names = check_names
+    )
+  
+  attributes(updated_data)$group_DF <-
+    attributes(omicsData)$group_DF
   attributes(updated_data)$filters <- attributes(omicsData)$filters
-  attributes(updated_data)$meta_info <- attributes(omicsData)$meta_info
+  attributes(updated_data)$meta_info <-
+    attributes(omicsData)$meta_info
+  
+  message(paste(
+    num_replaced,
+    "instances of",
+    x,
+    "have been replaced with",
+    y,
+    sep = " "
+  ))
+  return(updated_data)
+}
+
+
 
-  message(paste(num_replaced, "instances of", x, "have been replaced with", y, sep=" "))
+# function for nmrdata
+#' @name edata_replace
```

Comment:
Add @export call and roxygenize for nmrData use

Reply:
Thanks, done!